### DISCUSSION TITLE
Error using dynamic import for SSR False, TypeError: item is not a function

### DISCUSSION THREAD

**Discussion by blackmo18** [vercel/next.js#19977] [Help] [answered]
I am using Next Js v10.0.0, and regraph lastest trial sdk.

im a trying to import the following

`const neighbors = dynamic(() => import('regraph/analysis'), { ssr: false })`
since im having an error `navigator is not defined`

but when it using at runtime it says: `TypeError: neighbors is not a function`

someone knows how to use this?

would love to know the works around.

**Comment by jamesmosier** [comment, marked as answer]
Is `regraph/analysis` a React component? `next/dynamic` only works for React components. For other JavaScript functions, you can use "native" dynamic imports: https://nextjs.org/docs/advanced-features/dynamic-import

**Reply by dipunj** [reply]
By native imports do you mean

```tsx
const SDK = import('some-sdk-library');
```

inside the component, right?

**Reply by jamesmosier** [reply]
Yes, see the [docs I linked](https://nextjs.org/docs/advanced-features/dynamic-import) for an example.

```js
export default function Page() {
  const [results, setResults] = useState()

  return (
    <div>
      <input
        type="text"
        placeholder="Search"
        onChange={async (e) => {
          const { value } = e.currentTarget
          // Dynamically load fuse.js
          const Fuse = (await import('fuse.js')).default
          const fuse = new Fuse(names)

          setResults(fuse.search(value))
        }}
      />
      <pre>Results: {JSON.stringify(results, null, 2)}</pre>
    </div>
  )
}
```

[tc39 dynamic import proposal](https://github.com/tc39/proposal-dynamic-import)

**Reply by dipunj** [reply]
Hey @jamesmosier thanks for the help.

Really Sorry. I missed it when I was skimming through the doc, I was expecting it outside the JSX in the mentioned example.

Thanks again!

**Comment by salvatorericcardi** [comment]
Hi @jamesmosier,
I have the same issue but I can't understand it seems everything is ok. Maybe something goes wrong, but these are the codes block:

Where I use jsx.element
```
import dynamic from "next/dynamic"
const Map = dynamic(() => import("../../shared/map"), { ssr:false })
```

Where I declare jsx.element
```
import { LatLngExpression } from 'leaflet';
import { MapContainer, TileLayer, Marker, Popup } from 'react-leaflet';

export default function Map(): JSX.Element {
    let position: LatLngExpression = [41.3576, 13.4374]

    return <div id='map' className='col-lg-6 col-12' style={{height:'300px'}}>
        <MapContainer center={position} zoom={15} scrollWheelZoom={false} style={{height:'300px'}}>
            <TileLayer attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors' url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" />
            <Marker position={position}>
                <Popup>Stadio {"D. Purificato"}</Popup>
            </Marker>
        </MapContainer>
    </div>
}
```
